src/main/scala/com/scalapenos/riak/internal/RiakHttpClientHelper.scala

```diff
@@ -165,7 +164,9 @@ private[riak] class RiakHttpClientHelper(system: ActorSystem) extends RiakUriSup
 
     val entity = JsObject("props" -> JsObject(newProperties.map(property ⇒ (property.name -> property.json)).toMap))
 
-    httpRequest(Put(PropertiesUri(server, bucket), entity)).map { response ⇒
+    // For some reason, Riak set bucket props HTTP endpoint doesn't handle compressed request properly.
+    // So we disable compression for this request unconditionally.
```

Comment:
yep. Done. It's this one: https://github.com/agemooij/riak-scala-client/issues/41
we have also found another interesting issue: https://github.com/agemooij/riak-scala-client/issues/42